Description: Set define for calling convention only for Win32 targets.
 The compiler emitted a warning that it ignores the stdcall calling
 convention for targets other than Windows. This patch defines the
 NSISCALL only for Win32 targets.
Author: Thomas Gaugler <thomas@dadie.net>

--- a/SCons/Config/gnu
+++ b/SCons/Config/gnu
@@ -9,6 +9,14 @@
 ### cross compiling
 
 def cross_env(env):
+	# Set stdcall calling convention for Win32 targets
+	if env.has_key('CPPDEFINES'):
+		for i in range(len(env['CPPDEFINES'])):
+			if 'NSISCALL' in env['CPPDEFINES'][i]:
+				env['CPPDEFINES'][i] = ('NSISCALL', '$STDCALL')
+				break
+	else:
+		env.Append(CPPDEFINES = [('NSISCALL', '$STDCALL')])
 	if env['PLATFORM'] != 'win32':
 		env.Tool('crossmingw', toolpath = [Dir('../Tools').rdir()])
 
@@ -35,7 +43,7 @@
 
 ### defines
 
-defenv.Append(CPPDEFINES = [('NSISCALL', '$STDCALL')])
+defenv.Append(CPPDEFINES = [('NSISCALL', '')])
 
 ### helper functions
 
